File: Scripts_Archive/get_funcs.js
const fs = require('fs');

const content = fs.readFileSync('js/main.js', 'utf8');
const lines = content.split('\n');

let start = -1;
for (let i = 0; i < lines.length; i++) {
    if (lines[i].includes('function createMsgElement(')) {
        start = i;
        break;
    }
}

if (start !== -1) {
    let braces = 0;
    let started = false;
    let end = start;
    for (let i = start; i < lines.length; i++) {
        let line = lines[i];
        if (line.includes('{')) {
            braces += (line.match(/\{/g) || []).length;
            started = true;
        }
        if (line.includes('}')) {
            braces -= (line.match(/\}/g) || []).length;
        }
        if (started && braces === 0) {
            end = i;
            break;
        }
    }
    console.log(lines.slice(start, end + 1).join('\n'));
} else {
    console.log("Not found");
}

File: Scripts_Archive/get_funcs.py
import re

with open('js/main.js', 'r', encoding='utf-8') as f:
    content = f.read()

# simple regex to find createMsgElement function
match = re.search(r'function createMsgElement\([^)]*\)\s*\{(?:[^{}]*|\{(?:[^{}]*|\{[^{}]*\})*\})*\}', content)
if match:
    print(match.group(0))
else:
    print("Not found by simple regex, extracting line range")
    lines = content.split('\n')
    start = -1
    for i, line in enumerate(lines):
        if 'function createMsgElement(' in line:
            start = i
            break
    if start != -1:
        end = start
        braces = 0
        started = False
        for i in range(start, len(lines)):
            line = lines[i]
            if '{' in line:
                braces += line.count('{')
                started = True
            if '}' in line:
                braces -= line.count('}')
            if started and braces == 0:
                end = i
                break
        print('\n'.join(lines[start:end+1]))


